### PR TITLE
[feat] Fetch Files From Canvas For Supabase

### PR DESCRIPTION
**Pull request by nathandtam**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Adding functionality for syncing files to Supabase bucket
- File metadata goes to Files table and downloads go to lesson-files bucket
- Changed storage_path in Supabase to be unique

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2025-10-26 at 2 18 48 PM
Screenshot 2025-10-26 at 2 19 02 PM

## 👍 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 👍 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://developerdocs.instructure.com/services/canvas/resources/files

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @nehaahussain

### CODE REVIEW

File: api/canvas/queries/query.ts [hunk 1]
```diff
@@ -63,3 +63,56 @@ export async function fetchAllModules(
...
+        Accept: "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch files: ${response.status}`);
+    }
+
+    const files = await response.json();
+    console.log(files);
```
Comment (kylezryr) [inline api/canvas/queries/query.ts]: you should remove this console.log

**Review by kylezryr — APPROVED** [review]
looking good nathan! no comments you are the goat 🐐  just remove a console log and we r good 🤩

**Review by nehaahussain — APPROVED** [review]
LGTM GREAT JOB NATHAN! 

next steps:
these functions may insert files (or other data like modules) before the related lessons exist which causes a foreign key violation. we need to ensure lessons are inserted first.
